Rework of the rsyslog formula

elemoine opened the pull request:
This PR fixes a few things in the formula, and adds support for Heka.

Commit:
Correctly handle "enabled" for config elements

Commit:
Better formatting of config file

Commit:
Use $Template rather than $template in config file

Commit:
Use RFC3164 log format by default

Commit:
Disable xconsole output by default

Commit:
Support Heka log_collector

Commit:
Use /var/log/auth.log and kern.log

Commit:
Remove files/sensu.conf

fpytloun commented:
Yes, I like this approach - much better than the one with linux formula.
Whole functionality needs to be tested anyway to verify status of this formula.

Commit:
Configure Heka logstreamer based on output config

elemoine commented:
> Yes, I like this approach - much better than the one with linux formula.
> Whole functionality needs to be tested anyway to verify status of this formula.

Thanks for the review. I'll let you know when this is ready for final review.

Commit:
Do not purge /etc/rsyslog.d

Remove the default /etc/rsyslog.d/50-default.conf config file created by
the rsyslog package instead of completely purging the /etc/rsyslog.d/
directory. Purging will remove all the config files added by other
packages (the haproxy package adds /etc/rsyslog.d/49-haproxy.conf for
example).

elemoine commented:
This is now ready for review. I'd recommend reviewing this commit by commit, instead of looking at the full diff.

Commit:
Collect syslog haproxy logs

simonpasquier commented on rsyslog/meta/heka.yml:
```diff
@@ -25,7 +25,14 @@ log_collector:
     {%- endfor %}
-  {%- endif %}
+    {%- endif %}
+    syslog_haproxy:
```
Naive question but shouldn't it be in the HAProxy formula?

simonpasquier commented:
Otherwise great work!
